File: scratch/remote_deploy.py
import paramiko
import time

host = "141.98.48.172"
user = "root"
password = "OCNtS3xPhbo4"

def run_remote_commands():
    try:
        print(f"🔗 Sunucuya bağlanılıyor: {host}...")
        ssh = paramiko.SSHClient()
        ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        ssh.connect(host, username=user, password=password, timeout=20)
        
        commands = [
            "cd /root/odelink-shop",
            "git fetch origin main",
            "git reset --hard origin/main",
            "npm install",
            "pm2 restart all",
            "echo '--- VDS RAPORU ---'",
            "df -h /",
            "free -h",
            "nproc",
            "lscpu | grep 'Model name'"
        ]
        
        full_command = " && ".join(commands)
        print("🚀 Komutlar gönderiliyor...")
        stdin, stdout, stderr = ssh.exec_command(full_command)
        
        output = stdout.read().decode()
        error = stderr.read().decode()
        
        if output:
            print("✅ ÇIKTI:\n", output)
        if error:
            print("⚠️ UYARI/HATA:\n", error)
            
        ssh.close()
        print("🏁 İşlem tamamlandı.")
        
    except Exception as e:
        print(f"❌ HATA: {str(e)}")

if __name__ == "__main__":
    run_remote_commands()

File: scratch/remote_deploy.js
const { Client } = require('ssh2');
const conn = new Client();

const host = "141.98.48.172";
const user = "root";
const password = "OCNtS3xPhbo4";

console.log(`🔗 Sunucuya bağlanılıyor: ${host}...`);

conn.on('ready', () => {
  console.log('✅ Bağlantı başarılı!');
  
  const commands = [
    "PROJECT_DIR=$(find /root -name \"odelink-shop\" -type d -not -path \"*/.*\" | head -n 1)",
    "echo \"📂 Proje klasörü: $PROJECT_DIR\"",
    "cd $PROJECT_DIR",
    "git fetch origin main",
    "git reset --hard origin/main",
    "echo \"🔥 Docker konteynerleri güncelleniyor...\"",
    "docker compose down || true",
    "docker compose up -d --build",
    "echo \"✅ Konteynerler ayağa kalktı!\"",
    "docker ps",
    "pm2 stop all && pm2 delete all || true",
    "echo \"--- SİSTEM DURUMU ---\"",
    "df -h /",
    "free -h"
  ].join(" && ");

  console.log("🚀 Enterprise Deployment başlatıldı...");
  
  conn.exec(commands, (err, stream) => {
    if (err) throw err;
    stream.on('close', (code, signal) => {
      console.log('🏁 İşlem tamamlandı. Çıkış kodu: ' + code);
      conn.end();
    }).on('data', (data) => {
      console.log('✅ ÇIKTI:\n' + data);
    }).stderr.on('data', (data) => {
      console.log('⚠️ UYARI/HATA:\n' + data);
    });
  });
}).connect({
  host: host,
  port: 4383,
  username: user,
  password: password,
  readyTimeout: 30000
});

conn.on('error', (err) => {
  console.error('❌ BAĞLANTI HATASI:', err.message);
});
